server/protocols/jmap-rfc-8621/src/test/scala/org/apache/james/jmap/rfc/RequestObjectTest.scala

```diff
@@ -0,0 +1,203 @@
+          |  "methodCalls": [
+          |    [ "Core/echo", {
+          |      "arg1": "arg1data",
+          |      "arg2": "arg2data"
+          |    }, "c1" ],
+          |    [ "Core/echo2", {
+          |      "arg3": "arg3data",
+          |      "arg4": "arg4data"
+          |    }, "c2" ]
+          |  ]
+          |}
+          |""".stripMargin) must be(
+        JsSuccess(RequestObject(
+          using = Seq(Capability("urn:ietf:params:jmap:core"), Capability("urn:ietf:params:jmap:core2")),
+          methodCalls = Seq(expectedInvocation1, expectedInvocation2))))
+    }
+  }
+
+  "Serialize RequestObject" must {
+    "succeed when write to string without CreatedIds" in {
+      val methodName: MethodName = MethodName("Core/echo")
+      val argument: Arguments = Arguments(Json.obj("arg1" -> "arg1data", "arg2" -> "arg2data"))
+      val methodCallId: MethodCallId = MethodCallId("c1")
+      val invocation: Invocation = Invocation(methodName, argument, methodCallId)
+
+      val requestObject: RequestObject = RequestObject(
+        using = Seq(Capability("urn:ietf:params:jmap:core"), Capability("urn:ietf:params:jmap:core2")),
+        methodCalls = Seq(invocation))
+      val expectedValue = Json.prettyPrint(Json.parse(
+        """
+          |{
+          |  "using": [ "urn:ietf:params:jmap:core", "urn:ietf:params:jmap:core2"],
+          |  "methodCalls": [
+          |    [ "Core/echo", {
+          |      "arg1": "arg1data",
+          |      "arg2": "arg2data"
+          |    }, "c1" ]
+          |  ]
+          |}
+          |""".stripMargin))
+
+      Json.prettyPrint(Json.toJson(requestObject)) must be(expectedValue)
+    }
+
+    "succeed when write to string with CreatedIds" in {
+      val methodName: MethodName = MethodName("Core/echo")
+      val argument: Arguments = Arguments(Json.obj("arg1" -> "arg1data", "arg2" -> "arg2data"))
+      val methodCallId: MethodCallId = MethodCallId("c1")
+      val invocation: Invocation = Invocation(methodName, argument, methodCallId)
+      val createdIds: CreatedIds = CreatedIds(Map(
+        ClientId(id) -> ServerId(id)
+      ))
+      val requestObject: RequestObject = RequestObject(
+        using = Seq(Capability("urn:ietf:params:jmap:core"), Capability("urn:ietf:params:jmap:core2")),
+        methodCalls = Seq(invocation),
+        Option(createdIds))
+
+      val expectedValue = Json.prettyPrint(Json.parse(
+        """
+          |{
+          |  "using": [ "urn:ietf:params:jmap:core", "urn:ietf:params:jmap:core2"],
+          |  "methodCalls": [
+          |    [ "Core/echo", {
+          |      "arg1": "arg1data",
+          |      "arg2": "arg2data"
+          |    }, "c1" ]
+          |  ],
+          |  "createdIds":{"aHR0cHM6Ly93d3cuYmFzZTY0ZW5jb2RlLm9yZy8":"aHR0cHM6Ly93d3cuYmFzZTY0ZW5jb2RlLm9yZy8"}
+          |}
+          |""".stripMargin))
+
+      Json.prettyPrint(Json.toJson(requestObject)) must be(expectedValue)
```

Comment:
`Json.toJson(requestObject) must equal Json.parse(expectedValue)`

(Here and on any Json asserts)